coralproject/talk
# Email Confirmation and Fixes

## What does this PR do?

As always, these PR's start small and grow.. big.

This covers a few outstanding issues, simplifies lots of flows.

1. This mainly adds support for email verification requirements to the backend.
    - Sends emails with token when enabled
    - Tests to verify functionality
2. Changes the way that CSRF tokens are issued, consumed, and managed to simplify the flow.
    - Removed CSRF tokens from the body, moved to header
    - Disabled CSRF protection on unit tests (still enabled on E2E tests)
3. Session updates
    - Fixes issue with logout destroying entire session instead of only the user passport instance 
4. Endpoint conformity
    - Endpoints were adjusted to conform to outlined contributing guidelines for PUT endpoints, namely, only returning a 204
    - Endpoints that conflicted with `/user/:user_id` wildcard routes were moved to a new `/account` endpoint.

## How do I test this PR?

1. Build the client code via `npm run build`
2. Run unit tests via `npm test`
3. Run the server via `npm start` and perform a visual walkthrough of features to ensure compatibility.
